### PR TITLE
Ai add authorization

### PR DESCRIPTION
**Pull request by aizatibraimova**

### CODE REVIEW

**Review by CarlosCastanedaDev — APPROVED** [review]
Nice job, Aizat!

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -58,13 +59,21 @@ def destroy
...
-      params.require(:comment).permit(:author_id, :photo_id, :body)
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  def is_an_authorized_user
+    @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+    if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)
+      redirect_back fallback_location: root_url, alert: "Not authorized"
```
Comment (aborov) [inline app/controllers/comments_controller.rb:71]: Great job checking if the user shouldn't have access to a given comment

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (aborov) [inline app/controllers/photos_controller.rb:3]: Nice way of keeping user away from changing other users' photos and restricting available actions

File: config/routes.rb [hunk 3]
```diff
@@ -4,14 +4,14 @@
   devise_for :users
   
   resources :comments
-  resources :follow_requests
-  resources :likes
-  resources :photos
+  resources :follow_requests, except: [:index, :show, :new, :edit]
+  resources :likes, only: [:create, :destroy]
+  resources :photos, except: [:index]
```
Comment (jniewiel) [inline config/routes.rb:9]: Nice job limiting the routes for your resources!

Comment (aborov) [inline config/routes.rb:9]: Awesome work on route-level restrictions!

**Review by aborov — APPROVED** [review]
Looks good, Aizat!

File: app/views/photos/_photo.html.erb [hunk 4]
```diff
@@ -7,18 +7,20 @@
     </h2>
 
     <div>
-      <%= link_to edit_photo_path(photo), class: "btn btn-link btn-sm text-muted" do %>
-        <i class="fas fa-edit fa-fw"></i>
-      <% end %>
+      <% if current_user == photo.owner %>
```
Comment (jniewiel) [inline app/views/photos/_photo.html.erb:10]: Good job limiting the edit and delete photo links to only the photo's user!

**Review by jniewiel — APPROVED** [review]
Great job!

**Pull request merged**